micro_ros_common_diagnostics/src/hwmonitor.c

```diff
@@ -55,11 +53,11 @@ int main(int argc, const char * argv[])
   }
 
   // create rcl_node
-  rcl_node_t my_node = rcl_get_zero_initialized_node();
+  rcl_node_t updater_node = rcl_get_zero_initialized_node();
   rcl_node_options_t node_ops = rcl_node_get_default_options();
   rc = rcl_node_init(
-    &my_node,
-    UPDATER_NAME,
+    &updater_node,
+    "hwmonitor",
```

Comment:
Maybe I don't know the exact workflow you have in mind to use this module, but as a user, am I supposed to change the name of the node and the namespace?

And second, wouldn't it be more generic, to define the node name and the namespace with `#define`? That way, I could simply have a header that defines those macros, and here have a `#ifndef` guards

Reply:
`#define` or command line argument?

Reply:
Just to close this down a bit, IMHO I believe that if we provide it as a command-line argument, it's not meant to be used on an embedded system. I think all these arguments should come from a generator with a `.in` style. Or, simply hardcode them with `#define`.
Furthermore, as this is meant to go into the microcontroller, the user should have full ability to set it up following their own code structure. This is a little bit different from regular ROS packages where you pretty much just let people run the executable.
Thus, what about providing this `hwmonitor.c` as an example, instead of looking at it as **the** implementation?